drivers/bridge/bridge.go

```diff
@@ -876,6 +877,55 @@ func setHairpinMode(nlh *netlink.Handle, link netlink.Link, enable bool) error {
 	return nil
 }
 
+func waitForIfaces(ifaceNames []string, timeout time.Duration, trigger func() error) error {
+	// Map interface names to their readiness states (false by default)
+	ifaceStates := make(map[string]bool)
+	allRunning := func() bool {
+		for _, ifaceState := range ifaceStates {
+			if !ifaceState {
+				return false
+			}
+		}
+		return true
+	}
+	for _, ifaceName := range ifaceNames {
+		ifaceStates[ifaceName] = false
+	}
+
+	// Start watching for link changes
+	notifyLink := make(chan netlink.LinkUpdate)
+	notifyDone := make(chan struct{})
+	netlink.LinkSubscribe(notifyLink, notifyDone)
+	defer close(notifyDone)
+
+	// Run the given function to trigger link changes
+	if err := trigger(); err != nil {
+		return err
+	}
+
+	// Do not return until the given interfaces are running, or timeout
+	timer := time.After(timeout)
+	for {
+		select {
+		case update := <-notifyLink:
+			if update.Header.Type == syscall.RTM_NEWLINK &&
+				update.IfInfomsg.Flags&syscall.IFF_RUNNING != 0 {
```

Comment:
Okay, thanks, I've determined the difference is `systemd-networkd` is running and attempting to hijack the veth devices in our case.  I will look into this a bit more.